### DISCUSSION TITLE
"RuntimeError: The team cannot be saved while it is running." How to stop team so I can save team's state?

### DISCUSSION THREAD

**Discussion by hongfy197** [microsoft/autogen#4663] [Q&A]
0.4.0.dev11:

I want to save the team state when my application shutdown so I can resume the task later when my application restarted. I call `team.save_state()` function but it return error below. How can I stop team and do ?

Error:
```
File "/Users/user/.pyenv/versions/3.12.2/lib/python3.12/site-packages/autogen_agentchat/teams/_group_chat/_base_group_chat.py", line 504, in save_state
    raise RuntimeError("The team cannot be saved while it is running.")
RuntimeError: The team cannot be saved while it is running.
```

**Comment by ekzhu** [comment]
Can you provide a code to reproduce? Because this indicates that your team is still running `run` or `run_stream`, perhaps you have not closed the async generator returned from `run_stream`? i.e., returning when still in `async for` context.

**Reply by hongfy197** [reply]
```
stream = team.run_stream()
async for message in stream:
    print(message)
```

You meant I need to break the loop of `async for`  before I can save state?

**Reply by ekzhu** [reply]
You need to finish the async generator before you can save the state. So don't return or break from the async for loop, loop until finishes. 


You can also cancel the run if you don't want to wait for all team to stop. 

Cancellation: https://microsoft.github.io/autogen/dev/reference/python/autogen_agentchat.teams.html#autogen_agentchat.teams.BaseGroupChat.run_stream
